source/examples/write-data.js

```diff
@@ -19,52 +19,51 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import { getSolidDataset, getThing } from "@inrupt/solid-client";
-
-const profileResource = await getSolidDataset(
-  "https://vincentt.inrupt.net/profile/card"
-);
-
-const profile = getThing(
-  profileResource,
-  "https://vincentt.inrupt.net/profile/card#me"
-);
-
 // BEGIN-EXAMPLE-WRITE-DATA
 
 import {
+  getSolidDataset,
+  getThing
   setStringNoLocale,
+  addStringNoLocale
   setThing,
   saveSolidDatasetAt,
 } from "@inrupt/solid-client";
-import { FOAF } from "@inrupt/vocab-common-rdf";
+import { FOAF, VCARD } from "@inrupt/vocab-common-rdf";
+
+// 1a. Start with an existing Thing (i.e., profile).
+// Note: Login code has been omitted for brevity. See the Prerequisite section above.
+// ...
 
-/*
-   Start with a previously fetched Thing (i.e. profile).
+const myDataset = await getSolidDataset( "https://docs-example.inrupt.net/profile/card", { fetch: session.fetch });
+const profile = getThing( myDataset, "https://docs-example.inrupt.net/profile/card#me");
 
-   Use setStringNoLocale to create a NEW Thing
-     (i.e., updatedProfile) with the updated name data.
+// 1b. Modify the thing; specifically, update the name and add two nicknames.
+// Note: solid-client functions do not modify objects passed in as arguments. 
+// Instead the functions return new objects with the modifications.
+// That is, setStringNoLocale and addStringNoLocale return a new Thing and
+// - profile remains unchanged and 
+// - updatedProfile is changed only because it is explicitly set to the object returned from addStringNoLocale.
 
-   The passed-in Thing (i.e. profile) is unmodified.
-*/
+let updatedProfile = setStringNoLocale(profile, VCARD.fn, "Docs Example");
+updatedProfile = addStringNoLocale(updatedProfile, FOAF.nick, "docs");
+updatedProfile = addStringNoLocale(updatedProfile, FOAF.nick, "example");
 
-const updatedProfile = setStringNoLocale(profile, FOAF.name, "Vincent");
+// 2. Update SolidDataset with the updated Thing (i.e., updatedProfile). 
+// Note::  solid-client functions do not modify objects passed in as arguments. 
```

Comment:
(Removed double colon)

```suggestion
// Note:  solid-client functions do not modify objects passed in as arguments. 
```